hypervisor/arch/x86/vmx.c

```diff
@@ -1391,14 +1391,15 @@ static void init_exec_ctrl(struct vcpu *vcpu)
 						value64);
 
 		if (is_vapic_intr_delivery_supported()) {
-			/* these fields are supported only on processors
-			 * that support the 1-setting of the "virtual-interrupt
-			 * delivery" VM-execution control
+			/* Disable all EOI VMEXIT by default and
+			 * clear RVI and SVI.
 			 */
-			exec_vmwrite64(VMX_EOI_EXIT0_FULL, ~0UL);
-			exec_vmwrite64(VMX_EOI_EXIT1_FULL, ~0UL);
-			exec_vmwrite64(VMX_EOI_EXIT2_FULL, ~0UL);
-			exec_vmwrite64(VMX_EOI_EXIT3_FULL, ~0UL);
+			exec_vmwrite64(VMX_EOI_EXIT0_FULL, 0UL);
+			exec_vmwrite64(VMX_EOI_EXIT1_FULL, 0UL);
+			exec_vmwrite64(VMX_EOI_EXIT2_FULL, 0UL);
+			exec_vmwrite64(VMX_EOI_EXIT3_FULL, 0UL);
+
+			exec_vmwrite16(VMX_GUEST_INTR_STATUS, 0);
```

Comment:
By looking at the name of the HV. We have is_vapic_supported(), and vapic to mean the VMX HW VAPIC feature. In the meantime, we have virtual lapic/vlapic.  I think we don't intend to distinguish them by with or without "l"  (sometime apic means ioapic + lapic, sometime it means lapic to me).   So, I intend to use vmx_vapic to stand for the hardware VMX VAPIC features, while remain vlapic for HV lapic device model. Same for vioapic.  How do you think?

Reply:
Agree. It will be more clear. Followed your suggestion:
vlapic: lapic device model(pure software solution)
vioapic: ioapic device model(pure software solution)
vmx_vapic: vmx apic(hardware solution)

And one more thing, the vmx_vapic and vlapic are located in the vlapic.c now, I am consider if need to separated as three files: virtual_lapic_core.c, vmx_vapic.c, vlapic.c. Are you prefer to the current architecture or this new one? Or other suggestions?

Reply:
Split the contents to a separate file is better... probably just vlapic.c & vmx_apic.c.. The former one means the basic implementation of lapic DM with full contents, the later one is the acceleration part using HW features.  This can be a next patch...

Reply:
ok. The reason to add virtual_lapic_core.c is due to some common code may need to put in it.. Anyway, I will try to split them first. We can discuss more in the 2nd patch set.